lib/Dodawanie_kolumny_z_epokami.py

```diff
@@ -0,0 +1,19 @@
+from pyspark.sql.functions import col, when
+from pyspark.sql.dataframe import DataFrame
+
+def add_epoch_column(df: DataFrame) -> DataFrame:
+  periods = [1901,1918,1926,1939,1954,1970,1985,1994,2009]
+  df_no_N = df.filter(df.startYear != "\\N")
```

Comment:
Tak się nie powinno tego robić. Albo dać assert, że takich rekordów nie ma. Albo wygenerować dla nich cechę, np. dać im osobną klasę.

Reply:
Podanie tabeli zamiast kolumny w przypadku sparka czy pandasa nie spowoduje wydłużenia działania. One nie generują kopii. Natomiast faktycznie jeśli przetwarzamy jedną kolumnę to przekazanie całego df może sugerować, że pracujemy na nim szerzej.

Tu jednak mamy po prostu pysparka. Z połączenia immutable, braku klasy takiej jak pandas.Series wychodzi, że właśnie tak należy zrobić. 

Na pewno natomiast nie wynika z nazwy funkcji, że usunie nam Nulle w tej tabeli i to jeszcze z użyciem hardcodowanej postaci tych nulli.

Reply:
Dodałem dwie asercje oraz zmieniłem listę periods na argument funkcji